### PR TITLE
de-jargonize, add/improve links, referential data

### PR DESCRIPTION
**Pull request by jennybc**
To explain why I mention XML:

I have never seen anything other than metadata come as or be stored as YAML, though I agree it's nice!

This section is more about how outside world or instrument will provide things, yes? So I think we have to acknowledge XML, though I try not to inflict it on myself.

Also I'm not sure "referential data" is self-explanatory.

### CODE REVIEW

File: index.md [hunk 1]
```diff
@@ -57,7 +57,7 @@ A practice is included in this minimal list if:
     *   And don't duplicate contents of stable, long-lived repositories (i.e., don't clone GenBank)
 2.  All synthesized data is stored in well-defined widely-used formats:
     *   CSV or HDF5 for tabular data
-    *   JSON or YAML for referential data
+    *   JSON, YAML, or XML for referential data, i.e. data that does not naturally form a rectangle
```
Comment (gvwilson) [inline index.md:60]: how about "networks and irregular data" rather than "referential data, i.e., ..."

File: index.md [hunk 2]
```diff
@@ -65,8 +65,8 @@ A practice is included in this minimal list if:
 4.  *Normalization* (the process of making data adhere to the rules in the preceding point) is treated as a processing step
     *   Raw data files are stored as they came
     *   Normalization steps are recorded textually in repeatable way
-5.  Filenames and directory names are semantically meaningful and structured to facilitate globbing
-    *   Files as field-field-field.type
+5.  Filenames and directory names are semantically meaningful and anticipate the need to list and filter them programmatically, e.g. via regular expressions or "globbing".
```
Comment (gvwilson) [inline index.md:68]: Sure - a lot of dejargonification is doing to have to take place as these bullets are translated into humanese.

File: index.md (hunk 1 above)
Comment (jennybc) [inline index.md:60]: Yes those are good and mean more to me than "referential".

I also talk about this a lot as "nested lists" but I have lost all perspective. Would that mean anything to other people?

"Not a natural rectangle" is the most important bit, I suspect.

Comment (gvwilson) [inline index.md:60]: How about "not tabular" (I suspect non-programmers won't know what we 
mean by "rectangular" data)?

Comment (jennybc) [inline index.md:60]: Yes to "not tabular". Do you think "nested" means anything to a non-programmer?

Comment (gvwilson) [inline index.md:60]: Graphs (the node-and-arc kind) aren't really "nested"...

Comment (jennybc) [inline index.md:60]: Agree "nested" doesn't describe graphs. But that's not the only use of these formats. With "nested" I'm trying to evoke non-tabular data like this: [donuts.xml](http://adobe.github.io/Spry/data/donuts.xml).

**Pull request merged**